cmd/kubernetes-static/main.go

```diff
@@ -145,31 +212,56 @@ func main() {
 	if err := integration.Publish(); err != nil {
 		logrus.Fatalf("Error while publishing: %v", err)
 	}
+
+	fmt.Println()
 }
 
-func startStaticMetricsServer(k8sMetricsVersion string) string {
+func startStaticMetricsServer(content embed.FS, k8sMetricsVersion string) string {
 	// This will allocate a random port
 	listener, err := net.Listen("tcp", "127.0.0.1:0")
 	if err != nil {
 		panic(err)
 	}
 
-	endpoint := fmt.Sprintf("http://localhost:%d", listener.Addr().(*net.TCPAddr).Port)
+	endpoint := fmt.Sprintf("http://%s", listener.Addr())
 	fmt.Println("Hosting Mock Metrics data on:", endpoint)
 
 	mux := http.NewServeMux()
 
-	dataDir := fmt.Sprintf("./cmd/kubernetes-static/data/%s", k8sMetricsVersion)
-
-	path, err := filepath.Abs(dataDir)
+	k8sContent, err := fs.Sub(content, filepath.Join("data", k8sMetricsVersion))
 	if err != nil {
-		log.Fatal(errors.New("cannot start server"))
+		panic(err)
```

Comment:
Why prefer panic to `log.Fatal(err)`? Just curious

Reply:
Ah, I'm not sure. Perhaps we can still use `log.Fatal` + fmt.Sprintf, so we can format the received error as well. Good point.

Reply:
Well, also the code above this use `panic()`. I changed both to use `log.Fatal` (which shouldn't be used at all BTW, generally one should favor returning `error` and doing `os.Exit()` in one place explicitly.